drivers/PS2.c

```
@@ -229,24 +230,20 @@ static void ProcessMouseData(uint8_t data) {
-                if (changed_buttons & mask) {
-                    if ((mouse.buttons & mask) && OnMouseButtonDown) {
-                        OnMouseButtonDown(mouse.x, mouse.y, i + 1);
-                    } else if (OnMouseButtonUp) {
-                        OnMouseButtonUp(mouse.x, mouse.y, i + 1);
-                    }
-                }
-            }
+    // Handle button presses
+    for (int i = 0; i < 3; i++) {
+        uint8_t mask = 1 << i;
+        if ((mouse.buttons & mask) && OnMouseButtonDown) {
```

Clang (14.0.6)

[warning] repeated branch in conditional chain

(bugprone-branch-clone)